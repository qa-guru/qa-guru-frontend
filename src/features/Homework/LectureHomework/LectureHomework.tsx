import React from "react";
import { Paper, Stack, Typography } from "@mui/material";
import { ILectureHomework } from "./LectureHomework.types";
import Content from "../../../shared/Content";

const style = {
  paper: { padding: "20px" },
};

const LectureHomework: React.FC<ILectureHomework> = ({ data }) => {
  const { lectureHomeWork } = data;

  return (
    <>
      <Typography pt="30px" variant="h4" mb="15px">
        Домашнее задание
      </Typography>
      <Paper sx={style.paper}>
        <Stack spacing={0.5}>
          <Content content={lectureHomeWork!} />
        </Stack>
      </Paper>
    </>
  );
};

export default LectureHomework;
