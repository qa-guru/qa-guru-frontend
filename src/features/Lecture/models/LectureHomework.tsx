import React from "react";
import { Box, Typography } from "@mui/material";
import { useLectureHomeWorkQuery } from "../../../generated/graphql";
import { useParams } from "react-router-dom";

const LectureHomework: React.FC = () => {
  const { lessonId } = useParams();
  const { data } = useLectureHomeWorkQuery({
    variables: { lectureId: lessonId! },
  });

  return (
    <>
      <Typography variant="h6">Домашнее задание</Typography>
      <Box>{data?.lectureHomeWork}</Box>
    </>
  );
};

export default LectureHomework;
