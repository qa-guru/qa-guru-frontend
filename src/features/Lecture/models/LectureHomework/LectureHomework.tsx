import React from "react";
import { Box, TextField, Typography } from "@mui/material";
import { LoadingButton } from "@mui/lab";
import { LectureHomeWorkQuery } from "../../../../generated/graphql";
import DOMPurify from "isomorphic-dompurify";

interface ILectureHomework {
  data: LectureHomeWorkQuery;
}

const LectureHomework: React.FC<ILectureHomework> = ({ data }) => {
  const { lectureHomeWork } = data;

  const content = DOMPurify.sanitize(lectureHomeWork!);

  return (
    <>
      <Typography variant="h6">Домашнее задание</Typography>
      <Box pt="15px" pb="15px">
        <Typography
          variant="subtitle1"
          dangerouslySetInnerHTML={{ __html: content }}
        />
      </Box>
      <Typography variant="h6">Ваш ответ</Typography>
      <TextField
        multiline
        rows={5}
        placeholder="поле для ответа"
        variant="filled"
      />
      <Box textAlign="center">
        <LoadingButton sx={{ minWidth: "143px" }} variant="contained">
          Отправить
        </LoadingButton>
      </Box>
    </>
  );
};

export default LectureHomework;
