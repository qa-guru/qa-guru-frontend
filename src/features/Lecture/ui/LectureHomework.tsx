import React from "react";
import { Box, Link, Paper, Stack, Typography } from "@mui/material";
import ReactPlayer from "react-player";

interface ILectureHomework {
  contentLectureHomeWork: any;
}

const style = {
  paper: { padding: "20px" },
  box: { width: { xs: "100%", sm: "480px" } },
};

const LectureHomework: React.FC<ILectureHomework> = (props) => {
  const { contentLectureHomeWork } = props;

  return (
    <>
      <Typography pt="40px" variant="h4" mb="15px">
        Домашнее задание
      </Typography>
      <Paper sx={style.paper}>
        <Stack spacing={0.5}>
          {contentLectureHomeWork.map((item: any, index: any) => {
            switch (item.type) {
              case "title":
                return (
                  <Typography key={index} variant="h6">
                    {item.value}
                  </Typography>
                );
              case "text":
                return (
                  <Typography key={index} variant="subtitle1">
                    {item.value}
                  </Typography>
                );
              case "video":
                return (
                  <Box key={index} sx={style.box}>
                    <Typography variant="h6">{item.value}</Typography>
                    <ReactPlayer width="100%" height="270px" url={item.url} />
                  </Box>
                );
              case "link":
                return (
                  <Link key={index} underline="hover" href={item.url}>
                    {item.value}
                  </Link>
                );
            }
          })}
        </Stack>
      </Paper>
    </>
  );
};

export default LectureHomework;
