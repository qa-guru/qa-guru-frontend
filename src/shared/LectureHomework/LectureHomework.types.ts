import { LectureHomeWorkQuery } from "../../api/graphql/generated/graphql";

export interface ILectureHomework {
  dataLectureHomework: LectureHomeWorkQuery;
}
