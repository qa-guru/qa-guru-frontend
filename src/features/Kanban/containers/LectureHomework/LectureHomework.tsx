import React from "react";
import { useLectureHomeWorkQuery } from "../../../../api/graphql/generated/graphql";
import Spinner from "../../../../shared/components/Spinner";
import NoDataErrorMessage from "../../../../shared/components/NoDataErrorMessage";
import LectureHomework from "../../../../shared/LectureHomework";

interface ILectureHomeworkContainer {
  lectureId: string;
}

const LectureHomeworkContainer: React.FC<ILectureHomeworkContainer> = ({
  lectureId,
}) => {
  const { data: dataLectureHomework, loading: loadingLectureHomeWork } =
    useLectureHomeWorkQuery({
      variables: { lectureId: lectureId! },
    });

  if (loadingLectureHomeWork) return <Spinner />;

  if (!lectureId) return <NoDataErrorMessage />;

  return <LectureHomework dataLectureHomework={dataLectureHomework!} />;
};

export default LectureHomeworkContainer;
