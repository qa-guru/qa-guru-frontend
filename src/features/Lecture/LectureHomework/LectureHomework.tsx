import React from "react";
import { Box, Link, Paper, Stack, Typography } from "@mui/material";
import { ILectureHomework } from "./LectureHomework.types";

const mockLectureHomeWork = [
  {
    type: "text",
    value: "1. Зарегистрируйте аккаунт github.",
  },
  {
    type: "text",
    value: "2. Создайте новый репозиторий для домашнего задания.",
  },
  {
    type: "text",
    value: "3. Разработайте один автотест на проверку тестовой формы.",
  },
  {
    type: "link",
    value: "Тестовая форма demoqa.com",
    url: "https://demoqa.com/automation-practice-form",
  },
  {
    type: "text",
    value: "В поле ответа необходимо приложить ссылку на репозиторий.",
  },
];

const style = {
  paper: { padding: "20px" },
  box: { width: { xs: "100%", sm: "480px" } },
};

const LectureHomework: React.FC<ILectureHomework> = ({ data }) => {
  const { lectureHomeWork } = data;

  return (
    <Box>
      <Typography pt="40px" variant="h4" mb="15px">
        Домашнее задание
      </Typography>
      <Paper sx={style.paper}>
        <Stack spacing={0.5}>
          {mockLectureHomeWork.map((item: any, index: any) => {
            const { value, url, type } = item;

            switch (type) {
              case "title":
                return (
                  <Typography key={index} variant="h6">
                    {value}
                  </Typography>
                );
              case "text":
                return (
                  <Typography key={index} variant="subtitle1">
                    {value}
                  </Typography>
                );
              case "video":
                return (
                  <Box key={index}>
                    <Typography variant="h6">{value}</Typography>
                    <Box sx={style.box}>
                      <iframe
                        // @ts-ignore
                        style={style.iframe}
                        src={`https://www.youtube.com/embed/${url}`}
                        allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
                        allowFullScreen
                        title="Embedded youtube"
                      />
                    </Box>
                  </Box>
                );
              case "link":
                return (
                  <Link key={index} underline="hover" href={url}>
                    {value}
                  </Link>
                );
            }
          })}
        </Stack>
      </Paper>
    </Box>
  );
};

export default LectureHomework;
