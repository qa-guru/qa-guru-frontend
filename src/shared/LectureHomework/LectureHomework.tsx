import React from "react";
import { Paper, Stack, Typography } from "@mui/material";
import { ILectureHomework } from "./LectureHomework.types";
import { style } from "./styles";
import ContentSerialization from "../Serializers/ContentSerialization";

const LectureHomework: React.FC<ILectureHomework> = ({
  dataLectureHomework,
}) => {
  const { lectureHomeWork } = dataLectureHomework!;

  return (
    <>
      <Paper sx={style.paper}>
        <Typography mb={2} variant="h5">
          Домашнее задание
        </Typography>
        <Stack spacing={0.5}>
          <ContentSerialization content={lectureHomeWork!} />
        </Stack>
      </Paper>
    </>
  );
};

export default LectureHomework;
