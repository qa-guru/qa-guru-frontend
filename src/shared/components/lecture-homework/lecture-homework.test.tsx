import { MemoryRouter } from "react-router-dom";
import { render } from "test/utils-test";
import { lectureContentDto } from "shared/mocks/letcure-content-dto.mock";

import LectureHomework from "./lecture-homework";

describe("LectureHomework", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <LectureHomework lectureHomeWork={lectureContentDto} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
