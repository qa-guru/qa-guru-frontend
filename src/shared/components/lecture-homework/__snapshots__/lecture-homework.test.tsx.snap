// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LectureHomework > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1n8pi7-MuiPaper-root"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 css-1gr94hq-MuiTypography-root"
    >
      Домашнее задание
    </h5>
    <div
      class="MuiStack-root css-1yc1p5z-MuiStack-root"
    >
      <div
        class="MuiBox-root css-i6bazn"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-2sm3sf-MuiTypography-root"
        >
          Здесь будет отображаться текст домашнего задания к текущей лекции.
          <br />
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-2sm3sf-MuiTypography-root"
        >
          Пример домашнего задания:
          <br />
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-2sm3sf-MuiTypography-root"
        >
          1. Есть ли разница между $("h1 div"); и $("h1").$("div").
          <br />
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-2sm3sf-MuiTypography-root"
        >
          Может ли привести к тому что, поиск найдёт разные элементы?
          <br />
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-2sm3sf-MuiTypography-root"
        >
          2. Разработайте следующий автотест:
          <br />
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-2sm3sf-MuiTypography-root"
        >
           - Откройте страницу Selenide в Github
          <br />
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-2sm3sf-MuiTypography-root"
        >
           - Перейдите в раздел Wiki проекта
          <br />
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-2sm3sf-MuiTypography-root"
        >
           - Убедитесь, что в списке страниц (Pages) есть страница SoftAssertions
          <br />
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-2sm3sf-MuiTypography-root"
        >
           - Откройте страницу SoftAssertions, проверьте что внутри есть пример кода для JUnit5
          <br />
        </p>
      </div>
      <div
        class="MuiBox-root css-i6bazn"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-2sm3sf-MuiTypography-root"
        >
          Здесь будет отображаться текст домашнего задания к текущей лекции.
          <br />
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-2sm3sf-MuiTypography-root"
        >
          Пример домашнего задания:
          <br />
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-2sm3sf-MuiTypography-root"
        >
          1. Есть ли разница между $("h1 div"); и $("h1").$("div").
          <br />
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-2sm3sf-MuiTypography-root"
        >
          Может ли привести к тому что, поиск найдёт разные элементы?
          <br />
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-2sm3sf-MuiTypography-root"
        >
          2. Разработайте следующий автотест:
          <br />
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-2sm3sf-MuiTypography-root"
        >
           - Откройте страницу Selenide в Github
          <br />
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-2sm3sf-MuiTypography-root"
        >
           - Перейдите в раздел Wiki проекта
          <br />
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-2sm3sf-MuiTypography-root"
        >
           - Убедитесь, что в списке страниц (Pages) есть страница SoftAssertions
          <br />
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 css-2sm3sf-MuiTypography-root"
        >
           - Откройте страницу SoftAssertions, проверьте что внутри есть пример кода для JUnit5
          <br />
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
